tests/test-extensions/reconnection-ext/src/main/java/org/mule/extension/test/extension/reconnection/OtherReconnectionSource.java

```
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.test.extension.reconnection;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.connection.ConnectionProvider;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.scheduler.Scheduler;
+import org.mule.runtime.api.scheduler.SchedulerService;
+import org.mule.runtime.extension.api.annotation.param.Connection;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+import org.mule.runtime.extension.api.runtime.source.Source;
+import org.mule.runtime.extension.api.runtime.source.SourceCallback;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.inject.Inject;
+
+public class OtherReconnectionSource extends Source<Integer, Void> {
+
+  @Connection
+  ConnectionProvider<ReconnectableConnection> connectionProvider;
+
+  @Inject
+  SchedulerService schedulerService;
+
+  private final AtomicReference<ScheduledFuture<?>> scheduleWithFixedDelay = new AtomicReference<>();
+
+  private Scheduler scheduler;
+
+  private static final AtomicReference<Integer> countStartedSources = new AtomicReference<>(0);
+
+  private void delay(Long millis) {
+    try {
+      Thread.sleep(millis);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```

don't swallow this exception